flink-ecosystem-backend/src/main/java/org/apache/flink/community/FCPackageController.java

```diff
@@ -74,5 +85,85 @@ public String addPackage(@RequestBody FCPackage pack) {
 		packageService.savePackage(pack);
 		return "add package " + pack;
 	}
+
+	// endpoints for comments
+
+	@RequestMapping(value = "{pkg}/comments", method = RequestMethod.GET, produces = "application/json")
+	public String getComments(
+			@PathVariable("pkg") Integer pkgId,
+			@RequestParam Integer pageId) throws JsonProcessingException {
+		System.out.println("Come here" + pkgId + ":" + pageId);
+		List<Comment> comments = commentService.findCommentForPackage(pkgId);
+		int startIdx = (pageId - 1) * 25;
+		int endIdx = pageId * 25;
+		return new ObjectMapper().writeValueAsString(comments.subList(startIdx, endIdx));
+	}
+
+	@RequestMapping(value = "{pkg}/comments", method = RequestMethod.POST)
+	public void addComment(
+			@PathVariable("pkg") Integer pkgId,
+			@RequestBody Comment newComment) {
+		System.out.println("Add comment" + pkgId);
+		commentService.saveComment(newComment);
+	}
+
+	@RequestMapping(value = "{pkg}/comments/{comment_id}", method = RequestMethod.PATCH)
+	public void updateComment(
+			@PathVariable("pkg") Integer pkgId,
+			@PathVariable("comment_id") Integer cid,
+			@RequestBody Comment newComment) {
+		Comment old = commentService.findCommentById(cid);
+		commentService.saveComment(old.update(newComment));
+		System.out.println("update comment");
+
+	}
+
+	@RequestMapping(value = "{pkg}/comments/{comment_id}", method = RequestMethod.DELETE)
+	public void deleteComment(
+			@PathVariable("pkg") Integer pkgId,
+			@PathVariable("comment_id") Integer cid) {
+		Comment old = commentService.findCommentById(cid);
+		old.setValid(false);
+		commentService.saveComment(old);
+		System.out.println("del comment");
+	}
+
+
+	// endpoints for release
+	@RequestMapping(value = "{pkg}/releases", method = RequestMethod.GET, produces = "application/json")
+	public String getReleases() throws JsonProcessingException {
```

Comment:
http://localhost:2019/api/v1/packages/14656/releases
`could not prepare statement; nested exception is org.hibernate.exception.GenericJDBCException: could not prepare statement`

Reply:
will find out and fix this